A66-otel-stats.md

```diff
@@ -144,12 +144,53 @@ the scale to better fit the data.
 This section describes a `CallTracer` approach to collect the client and server
 per-attempt/call metrics. A CallTracer is a class that is instantiated for every
 call. This class has various methods that are invoked during the lifetime of the
-call. On the client side, the CallTracer knows about multiple attempts on the
+call. On the client-side, the CallTracer knows about multiple attempts on the
 same call, and creates a `CallAttemptTracer` object for each attempt, and the
-`CallAttemptTracer` gets invoked during the lifetime of the attempt.
+`CallAttemptTracer` gets invoked during the lifetime of the attempt. On the
+server-side, we have an equivalent `ServerCallTracer`. (There is no concept of
+an attempt on the server-side.)
 
 The OTel plugin will basically be a way of configuring CallTracer factories on
-gRPC clients and servers.
+gRPC channels and servers.
+
+A CallTracer needs to know the channel's target in the canonical form, and the
+full qualified method name for filling in the attributes needed on the metrics.
+Similarly on the server-side, the `ServerCallTracer` needs to know the method of
+the incoming call. Depending on the implementation details, the method may be
+propagated as part of the initial metadata.
+
+The following call-outs are needed on the `CallTracer` -
+
+*   When the call has been created. This call-out should be before payload
+    serialization.
+*   When new attempts are created on the call along with information on whether
+    the attempt was a transparent retry or not. (Attempts are created after name
+    resolution but before the pick.) This is also when it's expected for the
+    `CallAttemptTracer` to be created.
+*   When an attempt ends. This will be needed future stats around retries and
+    hedging. This information can also be propagated through the
+    `CallAttemptTracer` if the `CallAttemptTracer` keeps a reference to the
+    parent `CallTracer` object.
+*   When the call ends. This along with the call creation call-out allows the
+    `CallTracer` to calculate the call duration.
+
+The following call-outs are needed on the `CallAttemptTracer` -
```

Comment:
Do we want to say that all of these should be as close to the transport as possible?

Reply:
The position of the callout matters the most for the start and end timestamp, which for a client-side attempt is at the attempt top and not close to the transport.

The position for recording when a new message is sent/received does not matter for stats.

Reply:
> The position for recording when a new message is sent/received does not matter for stats.

That's probably fine for stats, but what about tracing and logging?  We'll be using this same API for tracing and logging, so if it matters for those, then we need to specify the requirements for this API correctly.

Reply:
Yeah, for tracing, I was hoping the tracing gRPC makes it more clear, but we would actually want two callouts. One for the compressed size, and one for the non-compressed size, and this would happen at the point of message compression.

Reply:
Why would we want two separate call-outs for tracing?

I think we would still want the call-out to be as close to the transport as possible, because even for tracing, it's important to reflect timing.  Someone looking at a trace may want to see how long it takes messages to be sent or received, and how much of that time is in the network vs. in gRPC itself.

Reply:
As discussed, I think this is something that needs to be covered by the tracing grfc